In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_excel('C:/Users/analysis/Desktop/SSAFY/projects/A_projects/pjt_final/movie.xlsx')

In [3]:
df

title          vote                   genre country  \
0           아무도 살아서 나갈 수 없다     관람객 평점 없음          ['미스터리', '공포']      영국   
1              드림걸: 수상한 콜센터     관람객 평점 없음                 ['코미디']      인도   
2                        트립     관람객 평점 없음           ['공포', '코미디']    노르웨이   
3     바다 탐험대 옥토넛 : 육지수호 대작전  관람객 평점 6.00점   ['애니메이션', '액션', '모험']      영국   
4                   왕십리 김종분     관람객 평점 없음               ['다큐멘터리']      한국   
...                     ...           ...                     ...     ...   
1807                 유혹의 계절     관람객 평점 없음       ['멜로/로맨스', '드라마']    아일랜드   
1808                  해치지않아  관람객 평점 7.32점                 ['코미디']      한국   
1809                 와일드 로즈  관람객 평점 8.90점                 ['드라마']      영국   
1810           블루버드 인 마이 하트     관람객 평점 없음     ['범죄', '액션', '드라마']     프랑스   
1811           박물관이 진짜 살아있다  관람객 평점 7.78점  ['애니메이션', '모험', '코미디']      체코   

      time   open_date                           actor     grade  \
0      86분      ['미상']                      ['마크 멘차카']  청소년 관람불가   
1     125분      ['미상']          ['아유쉬만 커라나', '비자이 라즈']   12세 관람가   
2     113분      ['미상']  ['누미 라파스', '엑셀 헨니', '아틀레 안톤센']  청소년 관람불가   
3      60분  2021.10.27           ['하성용', '정재헌', '엄상현']    전체 관람가   
4     102분  2021.11.11           ['김종분', '김귀임', '김종수']   12세 관람가   
...    ...         ...                             ...       ...   
1807  103분  2018.11.01  ['킬리언 머피', '캐서린 워커', '앤드류 스캇']  청소년 관람불가   
1808  117분  2020.01.15           ['안재홍', '강소라', '박영규']   12세 관람가   
1809  100분  2019.10.17            ['제시 버클리', '줄리 월터스']   15세 관람가   
1810   89분  2021.03.15   ['로랜드 몰러', '로라 르 란', '벨 배턴스']  청소년 관람불가   
1811   86분  2018.11.08           ['장은숙', '최원형', '장경희']    전체 관람가   

                                            poster_path  \
0     https://movie-phinf.pstatic.net/20211012_193/1...   
1     https://movie-phinf.pstatic.net/20210929_200/1...   
2     https://movie-phinf.pstatic.net/20211110_292/1...   
3     https://movie-phinf.pstatic.net/20210924_146/1...   
4     https://movie-phinf.pstatic.net/20211026_135/1...   
...                                                 ...   
1807  https://movie-phinf.pstatic.net/20181016_189/1...   
1808  https://movie-phinf.pstatic.net/20200115_223/1...   
1809  https://movie-phinf.pstatic.net/20191002_88/15...   
1810  https://movie-phinf.pstatic.net/20210310_84/16...   
1811  https://movie-phinf.pstatic.net/20181108_98/15...   

                                                content  
0     ['절박한 심정으로 미국에 밀입국한 멕시코 여성. 허름한 클리블랜드 하숙집에서 묵으...  
1     ['카람은 여자 목소리를 내는 데 천부적인 재능이 있는 남자로, 적절한 직업을 찾던...  
2     ['분노만 남은 결혼 생활. 부부는 서로를 죽이려는 계획을 숨기고 외딴 산장으로 여...  
3     ['바다 폭풍에 휩쓸린 옥토넛은 우연히 사막에 상륙한다.\r', <br/>, '\x...  
4     ['50년 넘게 노점을 해온 팔순의 김종분.', <br/>, '\xa0왕십리역 11...  
...                                                 ...  
1807  ['폭력적인 남편을 피해 아이들과 함께 친구 다니엘의 집을 찾은 ‘이반’\r', <...  
1808  ['생계형 수습 변호사 ‘태수’에게 찾아온 일생일대의 기회, 위기의 동물원 ‘동산파...  
1809  ['타고난 가창력으로 스코틀랜드 글래스고의 한 바에서 인기 몰이 중인 로즈의 꿈은 ...  
1810  ['교도소에서 출소한 대니는 로랑스와 클라라 모녀가 운영하는 낡은 모텔에서 새 삶을...  
1811  ['게임을 좋아하는 12살 소년 ‘하비’.', <br/>, '\xa0아버지가 일하는...  

[1812 rows x 10 columns]

In [4]:
df.astype({'title':'str'})

title          vote                   genre country  \
0           아무도 살아서 나갈 수 없다     관람객 평점 없음          ['미스터리', '공포']      영국   
1              드림걸: 수상한 콜센터     관람객 평점 없음                 ['코미디']      인도   
2                        트립     관람객 평점 없음           ['공포', '코미디']    노르웨이   
3     바다 탐험대 옥토넛 : 육지수호 대작전  관람객 평점 6.00점   ['애니메이션', '액션', '모험']      영국   
4                   왕십리 김종분     관람객 평점 없음               ['다큐멘터리']      한국   
...                     ...           ...                     ...     ...   
1807                 유혹의 계절     관람객 평점 없음       ['멜로/로맨스', '드라마']    아일랜드   
1808                  해치지않아  관람객 평점 7.32점                 ['코미디']      한국   
1809                 와일드 로즈  관람객 평점 8.90점                 ['드라마']      영국   
1810           블루버드 인 마이 하트     관람객 평점 없음     ['범죄', '액션', '드라마']     프랑스   
1811           박물관이 진짜 살아있다  관람객 평점 7.78점  ['애니메이션', '모험', '코미디']      체코   

      time   open_date                           actor     grade  \
0      86분      ['미상']                      ['마크 멘차카']  청소년 관람불가   
1     125분      ['미상']          ['아유쉬만 커라나', '비자이 라즈']   12세 관람가   
2     113분      ['미상']  ['누미 라파스', '엑셀 헨니', '아틀레 안톤센']  청소년 관람불가   
3      60분  2021.10.27           ['하성용', '정재헌', '엄상현']    전체 관람가   
4     102분  2021.11.11           ['김종분', '김귀임', '김종수']   12세 관람가   
...    ...         ...                             ...       ...   
1807  103분  2018.11.01  ['킬리언 머피', '캐서린 워커', '앤드류 스캇']  청소년 관람불가   
1808  117분  2020.01.15           ['안재홍', '강소라', '박영규']   12세 관람가   
1809  100분  2019.10.17            ['제시 버클리', '줄리 월터스']   15세 관람가   
1810   89분  2021.03.15   ['로랜드 몰러', '로라 르 란', '벨 배턴스']  청소년 관람불가   
1811   86분  2018.11.08           ['장은숙', '최원형', '장경희']    전체 관람가   

                                            poster_path  \
0     https://movie-phinf.pstatic.net/20211012_193/1...   
1     https://movie-phinf.pstatic.net/20210929_200/1...   
2     https://movie-phinf.pstatic.net/20211110_292/1...   
3     https://movie-phinf.pstatic.net/20210924_146/1...   
4     https://movie-phinf.pstatic.net/20211026_135/1...   
...                                                 ...   
1807  https://movie-phinf.pstatic.net/20181016_189/1...   
1808  https://movie-phinf.pstatic.net/20200115_223/1...   
1809  https://movie-phinf.pstatic.net/20191002_88/15...   
1810  https://movie-phinf.pstatic.net/20210310_84/16...   
1811  https://movie-phinf.pstatic.net/20181108_98/15...   

                                                content  
0     ['절박한 심정으로 미국에 밀입국한 멕시코 여성. 허름한 클리블랜드 하숙집에서 묵으...  
1     ['카람은 여자 목소리를 내는 데 천부적인 재능이 있는 남자로, 적절한 직업을 찾던...  
2     ['분노만 남은 결혼 생활. 부부는 서로를 죽이려는 계획을 숨기고 외딴 산장으로 여...  
3     ['바다 폭풍에 휩쓸린 옥토넛은 우연히 사막에 상륙한다.\r', <br/>, '\x...  
4     ['50년 넘게 노점을 해온 팔순의 김종분.', <br/>, '\xa0왕십리역 11...  
...                                                 ...  
1807  ['폭력적인 남편을 피해 아이들과 함께 친구 다니엘의 집을 찾은 ‘이반’\r', <...  
1808  ['생계형 수습 변호사 ‘태수’에게 찾아온 일생일대의 기회, 위기의 동물원 ‘동산파...  
1809  ['타고난 가창력으로 스코틀랜드 글래스고의 한 바에서 인기 몰이 중인 로즈의 꿈은 ...  
1810  ['교도소에서 출소한 대니는 로랑스와 클라라 모녀가 운영하는 낡은 모텔에서 새 삶을...  
1811  ['게임을 좋아하는 12살 소년 ‘하비’.', <br/>, '\xa0아버지가 일하는...  

[1812 rows x 10 columns]

In [5]:
print(df.shape)
df.info()

(1812, 10)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1812 entries, 0 to 1811
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   title        1812 non-null   object
 1   vote         1812 non-null   object
 2   genre        1812 non-null   object
 3   country      1812 non-null   object
 4   time         1808 non-null   object
 5   open_date    1812 non-null   object
 6   actor        1812 non-null   object
 7   grade        1812 non-null   object
 8   poster_path  1812 non-null   object
 9   content      1812 non-null   object
dtypes: object(10)
memory usage: 141.7+ KB


1. content 없는 값 확인

In [6]:
length = len(df.iloc[:])
print(length)

1812


In [7]:
no_content = []

In [8]:
for i in range(length):
    if df.iloc[i, 9] == '[]':
        no_content.append(i)

In [9]:
print(no_content)
print(len(no_content))

[10, 54, 59, 185, 488, 502, 715, 716, 729, 843, 844, 911, 952, 1006, 1009, 1024, 1041, 1048, 1071, 1091, 1157, 1161, 1282, 1343, 1375, 1485, 1559, 1614, 1719, 1778]
30


2. 배우가 없는 영화 목록 확인

In [10]:
no_actor = []
length = len(df.iloc[:])
print(length)

1812


In [11]:
grade_all = df.iloc[35,6]
grade_12 = df.iloc[17,6]
grade_15 = df.iloc[5,6]
grade_18 = df.iloc[15,6]
print(grade_all, grade_12, grade_15, grade_18)

['전체 관람가'] ['12세 관람가'] ['15세 관람가'] ['청소년 관람불가']


In [12]:
for i in range(length):
    if df.iloc[i, 6] == grade_all or df.iloc[i, 6] == grade_12 or df.iloc[i, 6] == grade_15 or df.iloc[i, 6] == grade_18:
        no_actor.append(i)

In [13]:
print(no_actor)
print(len(no_actor))

[5, 15, 17, 23, 24, 31, 35, 45, 49, 51, 59, 65, 67, 76, 83, 94, 98, 103, 104, 114, 119, 121, 122, 126, 129, 131, 132, 134, 148, 151, 154, 159, 166, 167, 170, 179, 186, 193, 195, 198, 201, 213, 214, 215, 218, 220, 226, 227, 233, 239, 242, 246, 249, 253, 256, 263, 268, 271, 276, 280, 286, 299, 302, 308, 328, 346, 367, 377, 411, 429, 447, 452, 456, 460, 463, 476, 484, 489, 497, 502, 509, 511, 516, 532, 533, 545, 555, 574, 609, 627, 635, 638, 639, 650, 651, 655, 656, 658, 663, 665, 666, 667, 672, 683, 684, 685, 686, 687, 704, 727, 732, 744, 754, 755, 766, 771, 777, 794, 795, 798, 864, 865, 866, 867, 874, 875, 876, 937, 945, 954, 957, 968, 970, 971, 974, 975, 976, 1005, 1008, 1009, 1011, 1024, 1029, 1034, 1035, 1038, 1065, 1076, 1098, 1103, 1118, 1120, 1129, 1130, 1131, 1132, 1134, 1135, 1136, 1141, 1142, 1143, 1153, 1167, 1192, 1210, 1226, 1239, 1240, 1241, 1242, 1244, 1251, 1254, 1255, 1268, 1269, 1271, 1284, 1306, 1324, 1325, 1326, 1343, 1352, 1364, 1396, 1397, 1399, 1400, 1417, 1436, 14

총 1782개 중에서 배우가 없는 영화는 213개이다.
충분히 삭제해도 괜찮은듯

3. 등급이 없는 영화 확인 + 전처리

- grade에서 ['전체 관람가'] 를 '전체 관람가'로 모두 변경

In [14]:
print(df.iloc[7,7] == ['전체 관람가'])
print(df.iloc[3,7] == '전체 관람가')

False
True


In [15]:
print(grade_all, grade_12, grade_15, grade_18)

['전체 관람가'] ['12세 관람가'] ['15세 관람가'] ['청소년 관람불가']


In [16]:
for i in range(length):
    if df.iloc[i, 7] == grade_all:
        df.iloc[i, 7] = '전체 관람가'

In [17]:
df.iloc[:20, 7]

0     청소년 관람불가
1      12세 관람가
2     청소년 관람불가
3       전체 관람가
4      12세 관람가
5       전체 관람가
6      12세 관람가
7       전체 관람가
8     청소년 관람불가
9       전체 관람가
10     12세 관람가
11     15세 관람가
12     12세 관람가
13     12세 관람가
14     15세 관람가
15      전체 관람가
16     15세 관람가
17      전체 관람가
18     15세 관람가
19     15세 관람가
Name: grade, dtype: object

- grade가 없는 값들 모두 no_grade에 담기

In [18]:
no_grade = []

In [19]:
for i in range(length):
    if df.iloc[i, 7] != '전체 관람가' and df.iloc[i, 7] != '12세 관람가' and df.iloc[i, 7] != '15세 관람가' and df.iloc[i, 7] != '청소년 관람불가':
        no_grade.append(i)

In [20]:
print(no_grade)

[525, 717, 1054, 1236]


In [21]:
df

title          vote                   genre country  \
0           아무도 살아서 나갈 수 없다     관람객 평점 없음          ['미스터리', '공포']      영국   
1              드림걸: 수상한 콜센터     관람객 평점 없음                 ['코미디']      인도   
2                        트립     관람객 평점 없음           ['공포', '코미디']    노르웨이   
3     바다 탐험대 옥토넛 : 육지수호 대작전  관람객 평점 6.00점   ['애니메이션', '액션', '모험']      영국   
4                   왕십리 김종분     관람객 평점 없음               ['다큐멘터리']      한국   
...                     ...           ...                     ...     ...   
1807                 유혹의 계절     관람객 평점 없음       ['멜로/로맨스', '드라마']    아일랜드   
1808                  해치지않아  관람객 평점 7.32점                 ['코미디']      한국   
1809                 와일드 로즈  관람객 평점 8.90점                 ['드라마']      영국   
1810           블루버드 인 마이 하트     관람객 평점 없음     ['범죄', '액션', '드라마']     프랑스   
1811           박물관이 진짜 살아있다  관람객 평점 7.78점  ['애니메이션', '모험', '코미디']      체코   

      time   open_date                           actor     grade  \
0      86분      ['미상']                      ['마크 멘차카']  청소년 관람불가   
1     125분      ['미상']          ['아유쉬만 커라나', '비자이 라즈']   12세 관람가   
2     113분      ['미상']  ['누미 라파스', '엑셀 헨니', '아틀레 안톤센']  청소년 관람불가   
3      60분  2021.10.27           ['하성용', '정재헌', '엄상현']    전체 관람가   
4     102분  2021.11.11           ['김종분', '김귀임', '김종수']   12세 관람가   
...    ...         ...                             ...       ...   
1807  103분  2018.11.01  ['킬리언 머피', '캐서린 워커', '앤드류 스캇']  청소년 관람불가   
1808  117분  2020.01.15           ['안재홍', '강소라', '박영규']   12세 관람가   
1809  100분  2019.10.17            ['제시 버클리', '줄리 월터스']   15세 관람가   
1810   89분  2021.03.15   ['로랜드 몰러', '로라 르 란', '벨 배턴스']  청소년 관람불가   
1811   86분  2018.11.08           ['장은숙', '최원형', '장경희']    전체 관람가   

                                            poster_path  \
0     https://movie-phinf.pstatic.net/20211012_193/1...   
1     https://movie-phinf.pstatic.net/20210929_200/1...   
2     https://movie-phinf.pstatic.net/20211110_292/1...   
3     https://movie-phinf.pstatic.net/20210924_146/1...   
4     https://movie-phinf.pstatic.net/20211026_135/1...   
...                                                 ...   
1807  https://movie-phinf.pstatic.net/20181016_189/1...   
1808  https://movie-phinf.pstatic.net/20200115_223/1...   
1809  https://movie-phinf.pstatic.net/20191002_88/15...   
1810  https://movie-phinf.pstatic.net/20210310_84/16...   
1811  https://movie-phinf.pstatic.net/20181108_98/15...   

                                                content  
0     ['절박한 심정으로 미국에 밀입국한 멕시코 여성. 허름한 클리블랜드 하숙집에서 묵으...  
1     ['카람은 여자 목소리를 내는 데 천부적인 재능이 있는 남자로, 적절한 직업을 찾던...  
2     ['분노만 남은 결혼 생활. 부부는 서로를 죽이려는 계획을 숨기고 외딴 산장으로 여...  
3     ['바다 폭풍에 휩쓸린 옥토넛은 우연히 사막에 상륙한다.\r', <br/>, '\x...  
4     ['50년 넘게 노점을 해온 팔순의 김종분.', <br/>, '\xa0왕십리역 11...  
...                                                 ...  
1807  ['폭력적인 남편을 피해 아이들과 함께 친구 다니엘의 집을 찾은 ‘이반’\r', <...  
1808  ['생계형 수습 변호사 ‘태수’에게 찾아온 일생일대의 기회, 위기의 동물원 ‘동산파...  
1809  ['타고난 가창력으로 스코틀랜드 글래스고의 한 바에서 인기 몰이 중인 로즈의 꿈은 ...  
1810  ['교도소에서 출소한 대니는 로랑스와 클라라 모녀가 운영하는 낡은 모텔에서 새 삶을...  
1811  ['게임을 좋아하는 12살 소년 ‘하비’.', <br/>, '\xa0아버지가 일하는...  

[1812 rows x 10 columns]

In [22]:
no_open = []

In [23]:
df.iloc[0,5] == "['미상']"

True

In [24]:
for i in range(length):
    if df.iloc[i,5] == "['미상']":
        no_open.append(i)

In [25]:
print(no_open)

[0, 1, 2, 5, 6, 12, 19, 30, 31, 32, 36, 37, 38, 40, 46, 54, 56, 59, 81, 87, 93, 117, 118, 126, 129, 133, 135, 141, 146, 152, 159, 163, 168, 170, 172, 173, 174, 185, 189, 208, 233, 283, 302, 310, 332, 333, 334, 336, 351, 352, 355, 360, 361, 362, 363, 366, 367, 368, 376, 378, 381, 384, 386, 391, 402, 403, 413, 415, 417, 430, 431, 472, 478, 481, 484, 485, 487, 488, 493, 499, 502, 503, 511, 518, 520, 525, 531, 532, 546, 550, 551, 554, 568, 592, 610, 611, 615, 619, 620, 621, 623, 630, 631, 632, 633, 634, 635, 636, 637, 638, 639, 640, 641, 642, 643, 644, 645, 646, 647, 648, 649, 650, 651, 652, 653, 654, 655, 656, 657, 658, 659, 660, 661, 662, 663, 664, 665, 666, 667, 668, 669, 670, 671, 672, 673, 681, 682, 683, 684, 685, 686, 694, 695, 700, 708, 709, 710, 711, 712, 713, 714, 715, 716, 717, 718, 729, 732, 733, 734, 741, 756, 780, 790, 794, 795, 796, 797, 798, 809, 812, 813, 816, 817, 819, 834, 835, 836, 837, 838, 839, 840, 841, 843, 844, 845, 846, 852, 853, 863, 864, 865, 866, 867, 868, 870, 

In [26]:
grouped = df.groupby('content')
# print(grouped.count())

In [27]:
del_list = no_content + no_actor + no_grade + no_open
del_list = sorted(list(set(del_list)))
print(del_list)
print(len(del_list))

[0, 1, 2, 5, 6, 10, 12, 15, 17, 19, 23, 24, 30, 31, 32, 35, 36, 37, 38, 40, 45, 46, 49, 51, 54, 56, 59, 65, 67, 76, 81, 83, 87, 93, 94, 98, 103, 104, 114, 117, 118, 119, 121, 122, 126, 129, 131, 132, 133, 134, 135, 141, 146, 148, 151, 152, 154, 159, 163, 166, 167, 168, 170, 172, 173, 174, 179, 185, 186, 189, 193, 195, 198, 201, 208, 213, 214, 215, 218, 220, 226, 227, 233, 239, 242, 246, 249, 253, 256, 263, 268, 271, 276, 280, 283, 286, 299, 302, 308, 310, 328, 332, 333, 334, 336, 346, 351, 352, 355, 360, 361, 362, 363, 366, 367, 368, 376, 377, 378, 381, 384, 386, 391, 402, 403, 411, 413, 415, 417, 429, 430, 431, 447, 452, 456, 460, 463, 472, 476, 478, 481, 484, 485, 487, 488, 489, 493, 497, 499, 502, 503, 509, 511, 516, 518, 520, 525, 531, 532, 533, 545, 546, 550, 551, 554, 555, 568, 574, 592, 609, 610, 611, 615, 619, 620, 621, 623, 627, 630, 631, 632, 633, 634, 635, 636, 637, 638, 639, 640, 641, 642, 643, 644, 645, 646, 647, 648, 649, 650, 651, 652, 653, 654, 655, 656, 657, 658, 659, 

In [28]:
movie = df.drop(del_list)
movie = movie.reset_index(drop=True)
movie.iloc[:3]

title          vote                  genre country  time  \
0  바다 탐험대 옥토넛 : 육지수호 대작전  관람객 평점 6.00점  ['애니메이션', '액션', '모험']      영국   60분   
1                왕십리 김종분     관람객 평점 없음              ['다큐멘터리']      한국  102분   
2           결혼 후에 남겨진 것들     관람객 평점 없음      ['멜로/로맨스', '드라마']      미국   97분   

    open_date                  actor    grade  \
0  2021.10.27  ['하성용', '정재헌', '엄상현']   전체 관람가   
1  2021.11.11  ['김종분', '김귀임', '김종수']  12세 관람가   
2      2021.1    ['로스 맥콜', '앨리슨 밀러']   전체 관람가   

                                         poster_path  \
0  https://movie-phinf.pstatic.net/20210924_146/1...   
1  https://movie-phinf.pstatic.net/20211026_135/1...   
2  https://movie-phinf.pstatic.net/20210928_272/1...   

                                             content  
0  ['바다 폭풍에 휩쓸린 옥토넛은 우연히 사막에 상륙한다.\r', <br/>, '\x...  
1  ['50년 넘게 노점을 해온 팔순의 김종분.', <br/>, '\xa0왕십리역 11...  
2  ['10년 된 부부 라이언과 클로디아는 옛 추억을 떠올리며, 신혼여행지 이탈리아를 ...

In [29]:
length = len(movie.iloc[:])
print(length)

1317


In [30]:
print(movie.iloc[4,2][1:8])

'애니메이션'


In [31]:
[movie.iloc[4,2]]

["['애니메이션']"]

In [32]:
for i in range(length):
    genre_len = len(movie.iloc[i,2])
    movie.iloc[i,2] = movie.iloc[i,2][1:genre_len-1]

In [33]:
movie['genre'] = movie['genre'].apply(lambda x: x.replace("'", ""))

In [34]:
movie['genre'] = movie['genre'].apply(lambda x: x.split(","))

In [35]:
movie[['genre_1', 'genre_2', 'genre_3', 'genre_4', 'genre_5', 'genre_6']] = pd.DataFrame(movie.genre.tolist(), index=movie.index)
movie.drop(['genre_5', 'genre_6'], axis=1, inplace=True)
movie.head()

title          vote              genre country  time  \
0  바다 탐험대 옥토넛 : 육지수호 대작전  관람객 평점 6.00점  [애니메이션,  액션,  모험]      영국   60분   
1                왕십리 김종분     관람객 평점 없음            [다큐멘터리]      한국  102분   
2           결혼 후에 남겨진 것들     관람객 평점 없음     [멜로/로맨스,  드라마]      미국   97분   
3              더 하더 데이 폴  관람객 평점 6.56점          [액션,  모험]      미국  137분   
4                  고장난 론  관람객 평점 8.48점            [애니메이션]      미국  106분   

    open_date                                   actor     grade  \
0  2021.10.27                   ['하성용', '정재헌', '엄상현']    전체 관람가   
1  2021.11.11                   ['김종분', '김귀임', '김종수']   12세 관람가   
2      2021.1                     ['로스 맥콜', '앨리슨 밀러']    전체 관람가   
3  2021.10.20        ['조나단 메이저스', '재지 비츠', '이드리스 엘바']  청소년 관람불가   
4  2021.10.27  ['자흐 갈리피아나키스', '잭 딜런 그레이저', '올리비아 콜맨']    전체 관람가   

                                         poster_path  \
0  https://movie-phinf.pstatic.net/20210924_146/1...   
1  https://movie-phinf.pstatic.net/20211026_135/1...   
2  https://movie-phinf.pstatic.net/20210928_272/1...   
3  https://movie-phinf.pstatic.net/20211013_124/1...   
4  https://movie-phinf.pstatic.net/20211013_94/16...   

                                             content genre_1 genre_2 genre_3  \
0  ['바다 폭풍에 휩쓸린 옥토넛은 우연히 사막에 상륙한다.\r', <br/>, '\x...   애니메이션      액션      모험   
1  ['50년 넘게 노점을 해온 팔순의 김종분.', <br/>, '\xa0왕십리역 11...   다큐멘터리    None    None   
2  ['10년 된 부부 라이언과 클로디아는 옛 추억을 떠올리며, 신혼여행지 이탈리아를 ...  멜로/로맨스     드라마    None   
3  ['숙적 루푸스 벅이 출소한 사실을 알게 된 서부의 무법자 냇 러브가 미뤄뒀던 복수...      액션      모험    None   
4  ['비봇을 갖는 것이 유일한 소원인 소심한 소년 ‘바니’에게도', <br/>, '\...   애니메이션    None    None   

  genre_4  
0    None  
1    None  
2    None  
3    None  
4    None

In [36]:
movie['genre_4'].isnull()

0       True
1       True
2       True
3       True
4       True
        ... 
1312    True
1313    True
1314    True
1315    True
1316    True
Name: genre_4, Length: 1317, dtype: bool

In [37]:
movie['actor'] = movie['actor'].apply(lambda x: x.replace("[", ""))
movie['actor'] = movie['actor'].apply(lambda x: x.replace("]", ""))
movie['actor'] = movie['actor'].apply(lambda x: x.replace("'", ""))

In [38]:
movie['actor'] = movie['actor'].apply(lambda x: x.split(","))

In [39]:
movie[['actor_1', 'actor_2', 'actor_3']] = pd.DataFrame(movie.actor.tolist(), index=movie.index)

In [40]:
movie

title          vote               genre country  time  \
0     바다 탐험대 옥토넛 : 육지수호 대작전  관람객 평점 6.00점   [애니메이션,  액션,  모험]      영국   60분   
1                   왕십리 김종분     관람객 평점 없음             [다큐멘터리]      한국  102분   
2              결혼 후에 남겨진 것들     관람객 평점 없음      [멜로/로맨스,  드라마]      미국   97분   
3                 더 하더 데이 폴  관람객 평점 6.56점           [액션,  모험]      미국  137분   
4                     고장난 론  관람객 평점 8.48점             [애니메이션]      미국  106분   
...                     ...           ...                 ...     ...   ...   
1312                 유혹의 계절     관람객 평점 없음      [멜로/로맨스,  드라마]    아일랜드  103분   
1313                  해치지않아  관람객 평점 7.32점               [코미디]      한국  117분   
1314                 와일드 로즈  관람객 평점 8.90점               [드라마]      영국  100분   
1315           블루버드 인 마이 하트     관람객 평점 없음     [범죄,  액션,  드라마]     프랑스   89분   
1316           박물관이 진짜 살아있다  관람객 평점 7.78점  [애니메이션,  모험,  코미디]      체코   86분   

       open_date                               actor     grade  \
0     2021.10.27                   [하성용,  정재헌,  엄상현]    전체 관람가   
1     2021.11.11                   [김종분,  김귀임,  김종수]   12세 관람가   
2         2021.1                    [로스 맥콜,  앨리슨 밀러]    전체 관람가   
3     2021.10.20        [조나단 메이저스,  재지 비츠,  이드리스 엘바]  청소년 관람불가   
4     2021.10.27  [자흐 갈리피아나키스,  잭 딜런 그레이저,  올리비아 콜맨]    전체 관람가   
...          ...                                 ...       ...   
1312  2018.11.01          [킬리언 머피,  캐서린 워커,  앤드류 스캇]  청소년 관람불가   
1313  2020.01.15                   [안재홍,  강소라,  박영규]   12세 관람가   
1314  2019.10.17                   [제시 버클리,  줄리 월터스]   15세 관람가   
1315  2021.03.15           [로랜드 몰러,  로라 르 란,  벨 배턴스]  청소년 관람불가   
1316  2018.11.08                   [장은숙,  최원형,  장경희]    전체 관람가   

                                            poster_path  \
0     https://movie-phinf.pstatic.net/20210924_146/1...   
1     https://movie-phinf.pstatic.net/20211026_135/1...   
2     https://movie-phinf.pstatic.net/20210928_272/1...   
3     https://movie-phinf.pstatic.net/20211013_124/1...   
4     https://movie-phinf.pstatic.net/20211013_94/16...   
...                                                 ...   
1312  https://movie-phinf.pstatic.net/20181016_189/1...   
1313  https://movie-phinf.pstatic.net/20200115_223/1...   
1314  https://movie-phinf.pstatic.net/20191002_88/15...   
1315  https://movie-phinf.pstatic.net/20210310_84/16...   
1316  https://movie-phinf.pstatic.net/20181108_98/15...   

                                                content genre_1 genre_2  \
0     ['바다 폭풍에 휩쓸린 옥토넛은 우연히 사막에 상륙한다.\r', <br/>, '\x...   애니메이션      액션   
1     ['50년 넘게 노점을 해온 팔순의 김종분.', <br/>, '\xa0왕십리역 11...   다큐멘터리    None   
2     ['10년 된 부부 라이언과 클로디아는 옛 추억을 떠올리며, 신혼여행지 이탈리아를 ...  멜로/로맨스     드라마   
3     ['숙적 루푸스 벅이 출소한 사실을 알게 된 서부의 무법자 냇 러브가 미뤄뒀던 복수...      액션      모험   
4     ['비봇을 갖는 것이 유일한 소원인 소심한 소년 ‘바니’에게도', <br/>, '\...   애니메이션    None   
...                                                 ...     ...     ...   
1312  ['폭력적인 남편을 피해 아이들과 함께 친구 다니엘의 집을 찾은 ‘이반’\r', <...  멜로/로맨스     드라마   
1313  ['생계형 수습 변호사 ‘태수’에게 찾아온 일생일대의 기회, 위기의 동물원 ‘동산파...     코미디    None   
1314  ['타고난 가창력으로 스코틀랜드 글래스고의 한 바에서 인기 몰이 중인 로즈의 꿈은 ...     드라마    None   
1315  ['교도소에서 출소한 대니는 로랑스와 클라라 모녀가 운영하는 낡은 모텔에서 새 삶을...      범죄      액션   
1316  ['게임을 좋아하는 12살 소년 ‘하비’.', <br/>, '\xa0아버지가 일하는...   애니메이션      모험   

     genre_3 genre_4     actor_1     actor_2   actor_3  
0         모험    None         하성용         정재헌       엄상현  
1       None    None         김종분         김귀임       김종수  
2       None    None       로스 맥콜      앨리슨 밀러      None  
3       None    None    조나단 메이저스       재지 비츠   이드리스 엘바  
4       None    None  자흐 갈리피아나키스   잭 딜런 그레이저   올리비아 콜맨  
...      ...     ...         ...         ...       ...  
1312    None    None      킬리언 머피      캐서린 워커    앤드류 스캇  
1313    None    None         안재홍         강소라       박영규  
1314    None    None      제시 버클리      줄리 월터스      None  
1315     드라마    None      로랜드 몰러      로라 르 란     벨 배턴스  
1316     코미디    None         장은숙         최원

In [41]:
movie['open_date']

0       2021.10.27
1       2021.11.11
2           2021.1
3       2021.10.20
4       2021.10.27
           ...    
1312    2018.11.01
1313    2020.01.15
1314    2019.10.17
1315    2021.03.15
1316    2018.11.08
Name: open_date, Length: 1317, dtype: object

In [42]:
movie['open_date'] = movie['open_date'].apply(lambda x: None if x == "['미상']" else x)

In [43]:
movie = movie[[x for x in movie.columns if x != 'content'] + ['content']]

In [44]:
movie['content'] = movie['content'].apply(lambda x: x.replace(", '\\xa0", ""))
movie['content'] = movie['content'].apply(lambda x: x.replace(", \"\\xa0", ""))
movie['content'] = movie['content'].apply(lambda x: x.replace("\\r', ", ""))
movie['content'] = movie['content'].apply(lambda x: x.replace("\\r\"', ", ""))
movie['content'] = movie['content'].apply(lambda x: x.replace("\\r\"", ""))
movie['content'] = movie['content'].apply(lambda x: x.replace("['", ""))
movie['content'] = movie['content'].apply(lambda x: x.replace("']", ""))

In [45]:
movie.drop(['genre', 'actor'], axis=1, inplace=True)

In [46]:
movie

title          vote country  time   open_date     grade  \
0     바다 탐험대 옥토넛 : 육지수호 대작전  관람객 평점 6.00점      영국   60분  2021.10.27    전체 관람가   
1                   왕십리 김종분     관람객 평점 없음      한국  102분  2021.11.11   12세 관람가   
2              결혼 후에 남겨진 것들     관람객 평점 없음      미국   97분      2021.1    전체 관람가   
3                 더 하더 데이 폴  관람객 평점 6.56점      미국  137분  2021.10.20  청소년 관람불가   
4                     고장난 론  관람객 평점 8.48점      미국  106분  2021.10.27    전체 관람가   
...                     ...           ...     ...   ...         ...       ...   
1312                 유혹의 계절     관람객 평점 없음    아일랜드  103분  2018.11.01  청소년 관람불가   
1313                  해치지않아  관람객 평점 7.32점      한국  117분  2020.01.15   12세 관람가   
1314                 와일드 로즈  관람객 평점 8.90점      영국  100분  2019.10.17   15세 관람가   
1315           블루버드 인 마이 하트     관람객 평점 없음     프랑스   89분  2021.03.15  청소년 관람불가   
1316           박물관이 진짜 살아있다  관람객 평점 7.78점      체코   86분  2018.11.08    전체 관람가   

                                            poster_path genre_1 genre_2  \
0     https://movie-phinf.pstatic.net/20210924_146/1...   애니메이션      액션   
1     https://movie-phinf.pstatic.net/20211026_135/1...   다큐멘터리    None   
2     https://movie-phinf.pstatic.net/20210928_272/1...  멜로/로맨스     드라마   
3     https://movie-phinf.pstatic.net/20211013_124/1...      액션      모험   
4     https://movie-phinf.pstatic.net/20211013_94/16...   애니메이션    None   
...                                                 ...     ...     ...   
1312  https://movie-phinf.pstatic.net/20181016_189/1...  멜로/로맨스     드라마   
1313  https://movie-phinf.pstatic.net/20200115_223/1...     코미디    None   
1314  https://movie-phinf.pstatic.net/20191002_88/15...     드라마    None   
1315  https://movie-phinf.pstatic.net/20210310_84/16...      범죄      액션   
1316  https://movie-phinf.pstatic.net/20181108_98/15...   애니메이션      모험   

     genre_3 genre_4     actor_1     actor_2   actor_3  \
0         모험    None         하성용         정재헌       엄상현   
1       None    None         김종분         김귀임       김종수   
2       None    None       로스 맥콜      앨리슨 밀러      None   
3       None    None    조나단 메이저스       재지 비츠   이드리스 엘바   
4       None    None  자흐 갈리피아나키스   잭 딜런 그레이저   올리비아 콜맨   
...      ...     ...         ...         ...       ...   
1312    None    None      킬리언 머피      캐서린 워커    앤드류 스캇   
1313    None    None         안재홍         강소라       박영규   
1314    None    None      제시 버클리      줄리 월터스      None   
1315     드라마    None      로랜드 몰러      로라 르 란     벨 배턴스   
1316     코미디    None         장은숙         최원형       장경희   

                                                content  
0     바다 폭풍에 휩쓸린 옥토넛은 우연히 사막에 상륙한다.<br/>바다 너머 육지에도 다...  
1     50년 넘게 노점을 해온 팔순의 김종분.', <br/>왕십리역 11번 출구 터줏대감...  
2     10년 된 부부 라이언과 클로디아는 옛 추억을 떠올리며, 신혼여행지 이탈리아를 다시...  
3     숙적 루푸스 벅이 출소한 사실을 알게 된 서부의 무법자 냇 러브가 미뤄뒀던 복수를 ...  
4     비봇을 갖는 것이 유일한 소원인 소심한 소년 ‘바니’에게도', <br/>드디어 ‘론...  
...                                                 ...  
1312  폭력적인 남편을 피해 아이들과 함께 친구 다니엘의 집을 찾은 ‘이반’<br/>매일 ...  
1313  생계형 수습 변호사 ‘태수’에게 찾아온 일생일대의 기회, 위기의 동물원 ‘동산파크’...  
1314  타고난 가창력으로 스코틀랜드 글래스고의 한 바에서 인기 몰이 중인 로즈의 꿈은 그곳...  
1315  교도소에서 출소한 대니는 로랑스와 클라라 모녀가 운영하는 낡은 모텔에서 새 삶을 시...  
1316  게임을 좋아하는 12살 소년 ‘하비’.', <br/>아버지가 일하는 장난감 박물관이...  

[1317 rows x 15 columns]

관람객 평점 없음 데이터


관람객 평점 존재 데이터


두개의 데이터 생성

In [47]:
unvote_data = movie[movie.vote=='관람객 평점 없음']
vote_data = movie[movie.vote!='관람객 평점 없음']

In [116]:
unvote_data.to_csv('unvote_data.csv')
vote_data.to_csv('vote_data.csv')

In [48]:
unvote_data

title       vote country  time   open_date     grade  \
1          왕십리 김종분  관람객 평점 없음      한국  102분  2021.11.11   12세 관람가   
2     결혼 후에 남겨진 것들  관람객 평점 없음      미국   97분      2021.1    전체 관람가   
5              피어썸  관람객 평점 없음      한국   82분  2021.10.07   15세 관람가   
6     피치 오브 타임 극장판  관람객 평점 없음      한국   99분  2021.10.21   12세 관람가   
8            레드 아워  관람객 평점 없음      미국   93분      2021.1   15세 관람가   
...            ...        ...     ...   ...         ...       ...   
1300   로빈 후드: 리벨리언  관람객 평점 없음      영국   92분  2018.11.29   15세 관람가   
1301      엔젤 오브 마인  관람객 평점 없음      미국   98분  2019.10.30   15세 관람가   
1311           방문객  관람객 평점 없음      한국   75분  2019.03.27   15세 관람가   
1312        유혹의 계절  관람객 평점 없음    아일랜드  103분  2018.11.01  청소년 관람불가   
1315  블루버드 인 마이 하트  관람객 평점 없음     프랑스   89분  2021.03.15  청소년 관람불가   

                                            poster_path genre_1 genre_2  \
1     https://movie-phinf.pstatic.net/20211026_135/1...   다큐멘터리    None   
2     https://movie-phinf.pstatic.net/20210928_272/1...  멜로/로맨스     드라마   
5     https://movie-phinf.pstatic.net/20210917_247/1...      공포    미스터리   
6     https://movie-phinf.pstatic.net/20210930_277/1...  멜로/로맨스     드라마   
8     https://movie-phinf.pstatic.net/20210923_101/1...      액션     스릴러   
...                                                 ...     ...     ...   
1300  https://movie-phinf.pstatic.net/20181116_139/1...      액션      모험   
1301  https://movie-phinf.pstatic.net/20191011_108/1...     드라마     스릴러   
1311  https://movie-phinf.pstatic.net/20190315_217/1...     드라마    미스터리   
1312  https://movie-phinf.pstatic.net/20181016_189/1...  멜로/로맨스     드라마   
1315  https://movie-phinf.pstatic.net/20210310_84/16...      범죄      액션   

     genre_3 genre_4     actor_1       actor_2        actor_3  \
1       None    None         김종분           김귀임            김종수   
2       None    None       로스 맥콜        앨리슨 밀러           None   
5       None    None         조동혁           차선우           None   
6       None    None  지미 칸 크리사나판           최재현   토미 싯티촉 푸에크풀폴   
8       None    None     타이레스 깁슨        존 말코비치           None   
...      ...     ...         ...           ...            ...   
1300    None    None       벤 프리먼        마리 에버릿    제임스 올리버 휘트리   
1301    None    None      누미 라파스   이본느 스트라호브스키         루크 에반스   
1311     판타지    None          정호          None           None   
1312    None    None      킬리언 머피        캐서린 워커         앤드류 스캇   
1315     드라마    None      로랜드 몰러        로라 르 란          벨 배턴스   

                                                content  
1     50년 넘게 노점을 해온 팔순의 김종분.', <br/>왕십리역 11번 출구 터줏대감...  
2     10년 된 부부 라이언과 클로디아는 옛 추억을 떠올리며, 신혼여행지 이탈리아를 다시...  
5     공연을 할 수 없게 된 ‘승연’(조동혁)의 밴드는', <br/>재정비를 위해 춘천으...  
6     친구 윤오의 연락을 받고 한국에 도착한 피치.', <br/>그렇게 날 반갑게 맞아준...  
8     아동보호국에서 근무하는 카일은 아프가니스탄 파병지에서 동료를 잃었던 끔찍한 기억으로...  
...                                                 ...  
1300  십자군 원정에서 돌아온 로빈 후드.<br/>주장관의 폭정에 반기를 들고 마을 사람들...  
1301  7년 전 사고로 죽은 아이에 대한 그리움으로<br/>일, 가족 그리고 자기 자신마저...  
1311  죽어야 하는 오늘, 찾아오는 낯선 자들', <br/>탈북한 북의 고위 간부가 해외에...  
1312  폭력적인 남편을 피해 아이들과 함께 친구 다니엘의 집을 찾은 ‘이반’<br/>매일 ...  
1315  교도소에서 출소한 대니는 로랑스와 클라라 모녀가 운영하는 낡은 모텔에서 새 삶을 시...  

[581 rows x 15 columns]

In [54]:
gr = vote_data.groupby('grade')
print(gr.count())

          title  vote  country  time  open_date  poster_path  genre_1  \
grade                                                                   
12세 관람가     226   226      226   226        226          226      226   
15세 관람가     261   261      261   261        261          261      261   
전체 관람가      178   178      178   178        178          178      178   
청소년 관람불가     71    71       71    71         71           71       71   

          genre_2  genre_3  genre_4  actor_1  actor_2  actor_3  content  
grade                                                                    
12세 관람가       101       23        0      226      203      128      226  
15세 관람가       159       45        2      261      245      178      261  
전체 관람가         71       45        8      178      159      117      178  
청소년 관람불가       41       18        3       71       64       49       71  


In [49]:
vote_data

title          vote country  time   open_date     grade  \
0     바다 탐험대 옥토넛 : 육지수호 대작전  관람객 평점 6.00점      영국   60분  2021.10.27    전체 관람가   
3                 더 하더 데이 폴  관람객 평점 6.56점      미국  137분  2021.10.20  청소년 관람불가   
4                     고장난 론  관람객 평점 8.48점      미국  106분  2021.10.27    전체 관람가   
7                    가치 캅시다  관람객 평점 7.40점      한국   93분  2021.10.28   15세 관람가   
13                      F20  관람객 평점 9.50점      한국  104분  2021.10.06   15세 관람가   
...                     ...           ...     ...   ...         ...       ...   
1309                 어린 의뢰인  관람객 평점 9.14점      한국  114분  2019.05.22   12세 관람가   
1310               블라인드 멜로디  관람객 평점 8.54점      인도  139분  2019.08.28   15세 관람가   
1313                  해치지않아  관람객 평점 7.32점      한국  117분  2020.01.15   12세 관람가   
1314                 와일드 로즈  관람객 평점 8.90점      영국  100분  2019.10.17   15세 관람가   
1316           박물관이 진짜 살아있다  관람객 평점 7.78점      체코   86분  2018.11.08    전체 관람가   

                                            poster_path genre_1 genre_2  \
0     https://movie-phinf.pstatic.net/20210924_146/1...   애니메이션      액션   
3     https://movie-phinf.pstatic.net/20211013_124/1...      액션      모험   
4     https://movie-phinf.pstatic.net/20211013_94/16...   애니메이션    None   
7     https://movie-phinf.pstatic.net/20211012_41/16...     드라마    None   
13    https://movie-phinf.pstatic.net/20210928_86/16...     스릴러    None   
...                                                 ...     ...     ...   
1309  https://movie-phinf.pstatic.net/20190515_46/15...     드라마    None   
1310  https://movie-phinf.pstatic.net/20190812_30/15...     스릴러    None   
1313  https://movie-phinf.pstatic.net/20200115_223/1...     코미디    None   
1314  https://movie-phinf.pstatic.net/20191002_88/15...     드라마    None   
1316  https://movie-phinf.pstatic.net/20181108_98/15...   애니메이션      모험   

     genre_3 genre_4     actor_1     actor_2   actor_3  \
0         모험    None         하성용         정재헌       엄상현   
3       None    None    조나단 메이저스       재지 비츠   이드리스 엘바   
4       None    None  자흐 갈리피아나키스   잭 딜런 그레이저   올리비아 콜맨   
7       None    None         김기현         강형석       남지우   
13      None    None         장영남         김정영       김강민   
...      ...     ...         ...         ...       ...   
1309    None    None         이동휘          유선       최명빈   
1310    None    None    아유쉬만 커라나          타부    라디카 압테   
1313    None    None         안재홍         강소라       박영규   
1314    None    None      제시 버클리      줄리 월터스      None   
1316     코미디    None         장은숙         최원형       장경희   

                                                content  
0     바다 폭풍에 휩쓸린 옥토넛은 우연히 사막에 상륙한다.<br/>바다 너머 육지에도 다...  
3     숙적 루푸스 벅이 출소한 사실을 알게 된 서부의 무법자 냇 러브가 미뤄뒀던 복수를 ...  
4     비봇을 갖는 것이 유일한 소원인 소심한 소년 ‘바니’에게도', <br/>드디어 ‘론...  
7     대한민국에서 영어 좀 한다는 남자라면 혹하는, 일생에 단 한 번만 지원할 수 있는 ...  
13    어디에 내놔도 자랑스러운 아들을 둔 엄마 ‘애란’은', <br/>군 생활을 떠났던 ...  
...                                                 ...  
1309  당신에게 찾아온 뜨거운 질문!', <br/>“당신은 이 아이를 외면하시겠습니까?”'...  
1310  눈이 보이지 않는 피아니스트 행세를 하며 <br/>라이브 레스토랑에서 공연을 하던 ...  
1313  생계형 수습 변호사 ‘태수’에게 찾아온 일생일대의 기회, 위기의 동물원 ‘동산파크’...  
1314  타고난 가창력으로 스코틀랜드 글래스고의 한 바에서 인기 몰이 중인 로즈의 꿈은 그곳...  
1316  게임을 좋아하는 12살 소년 ‘하비’.', <br/>아버지가 일하는 장난감 박물관이...  

[736 rows x 15 columns]